pkg/agent/openflow/client_test.go

```diff
@@ -258,29 +268,986 @@ func TestConcurrentFlowInstallation(t *testing.T) {
 			var wg sync.WaitGroup
 			done := make(chan struct{})
 
-			for i := 0; i < 2; i++ {
-				wg.Add(1)
-				cacheKey := fmt.Sprintf(tc.cacheKeyFormat, i)
-				go func() {
-					defer wg.Done()
-					_, _ = tc.fn(ofClient, cacheKey) // in mock we trust
-				}()
-			}
-			go func() {
-				defer close(done)
-				wg.Wait()
-			}()
+			for i := 0; i < 2; i++ {
+				wg.Add(1)
+				cacheKey := fmt.Sprintf(tc.cacheKeyFormat, i)
+				go func() {
+					defer wg.Done()
+					_, _ = tc.fn(fc, cacheKey) // in mock we trust
+				}()
+			}
+			go func() {
+				defer close(done)
+				wg.Wait()
+			}()
+
+			select {
+			case <-time.After(time.Second):
+				close(timeoutCh)
+				t.Fatal("timeoutCh, maybe there are some deadlocks")
+			case <-done:
+				assert.True(t, concurrentCalls.Load().(bool))
+			}
+		})
+	}
+
+}
+
+func newFakeClient(mockOFEntryOperations *oftest.MockOFEntryOperations,
+	enableIPv4,
+	enableIPv6 bool,
+	nodeType config.NodeType,
+	trafficEncapMode config.TrafficEncapModeType,
+	options ...clientOptionsFn) *client {
+
+	o := &clientOptions{
+		enableProxy:           true,
+		enableAntreaPolicy:    true,
+		enableEgress:          true,
+		proxyAll:              true,
```

Comment:
could it default to false? then every config in test has same default value as the value in production. Developer would know they need to update the option only when they are testing a non-default option.

Reply:
Done